utils/s2n_fork_detection.h

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include <stdint.h>
+
+int s2n_get_fork_generation_number(uint64_t *return_fork_generation_number);
+int assert_madv_wipeonfork_is_supported(void);
+int assert_map_inherit_zero_is_supported(void);
+
+/* Use for testing only */
+void FOR_TESTING_ignore_wipeonfork_and_inherit_zero(void);
+void FOR_TESTING_ignore_pthread_atfork(void);
```

Comment:
also, would it be possible to return `S2N_RESULT` instead of `int`? it provides some nicer guarantees around the values returned (see [the documentation](https://github.com/aws/s2n-tls/blob/f0739e6ffa629c61af87f67f601a3634aeb206f5/utils/s2n_result.c#L17-L76)).